pkg/loop/internal/broker.go

```diff
@@ -99,13 +110,6 @@ func (b *brokerExt) serve(name string, register func(*grpc.Server), deps ...reso
 		return 0, resource{}, ErrConnAccept{Name: name, ID: id, Err: err}
 	}
 
-	var server *grpc.Server
-	if b.NewServer == nil {
-		server = grpc.NewServer()
-	} else {
-		server = b.NewServer(nil)
-	}
-	register(server)
 	go func() {
 		defer b.closeAll(deps...)
 		if err := server.Serve(lis); err != nil {
```

Comment:
is there a potential logical race here wrt logging after tests are completed (l116)?  `resource` output args doesn't wait for the two created goroutines above to close.

Reply:
I think we could get an error if the listener is blocked on accepting a connection, but to your point, fairly unlikely.

Reply:
I think it's worth patching. Do you want to open a PR?

Reply:
Sure - will do!